### PR TITLE
community: Support for Fetching Previous/Next Segments in VertexAISearchRetriever

### PR DESCRIPTION
**Pull request by lgesuellip**
## PR Description

Currently, the [VertexAISearchRetriever](https://github.com/langchain-ai/langchain-google/blob/6dfdf9b57aa1f99d9c598a97e5729adb278883cf/libs/community/langchain_google_community/vertex_ai_search.py#L212) from langchain_google_community does not support fetching previous and next segments of documents for unstructured data. This is crucial for us to improve the quality of the fetched documents for our LLM.

The Google Discovery Engine API, including its Python library google.cloud.discoveryengine_v1beta, directly supports fetching previous and next segments. Relevant documentation can be found [here](https://cloud.google.com/php/docs/reference/cloud-discoveryengine/latest/V1beta.SearchRequest.ContentSearchSpec.ExtractiveContentSpec).

So, I implemented in this PR:

- Support for fetching and mapping previous/next segments in VertexAISearchRetriever.
  - Currently, only one `segment_answer` or `extractive_answer` is returned per SearchResult, so I set both to the maximum allowed value of 1.
  - Incorporated additional useful configurations from the Google API in the `_get_content_spec_kwargs` method.
- Handling of segments in the `_convert_unstructured_search_response` method.

## Relevant issues

https://github.com/langchain-ai/langchain-google/issues/502

## Type

🆕 New Feature
🐛 Bug Fix

## Testing

Since the library doesn’t have tests to cover these cases, I tested it using my data in GCP, checking different configurations and verifying that both the answers and segments are correct.

```
vertex = VertexAISearchRetriever(
    project_id=project_id,
    data_store_id=data_store_id,
    max_documents=2,
    max_extractive_segment_count=1,
    num_previous_segments=3,
    num_next_segments=3,
    return_extractive_segment_score=True,
)
```

Thank you Team!

### CODE REVIEW

**Comment by lkuligin** [comment]
Thanks for your contribution! Could you extend the integration test, please?

**Comment by lgesuellip** [comment]
I plan to mock the `_create_search_request` method by using a JSON with a format similar to the API responses. This will cover `extractive_segments` and `extractive_answers`. Then, I will create test cases to cover the `_convert_unstructured_search_response` method to ensure correct processing of these segments and answers.
What do you think @lkuligin ?

**Comment by lgesuellip** [comment]
I just added the tests that cover different scenarios and configurations of the VertexAISearchRetriever, ensuring that it correctly handles various types of search responses and generates appropriate Document objects.
I don't know why the CI is failing, as the tests pass locally.
I would love your feedback @lkuligin  .

File: libs/community/tests/integration_tests/test_vertex_ai_search.py
Comment (lkuligin) [inline libs/community/tests/integration_tests/test_vertex_ai_search.py:166]: this sounds more like a unit test to me, should we move it to unit tests maybe?

Comment (lgesuellip) [reply libs/community/tests/integration_tests/test_vertex_ai_search.py]: Yes, I moved the unit tests to the corresponding folder and added integration tests that use segments. I tested everything locally, and the tests are working well. Thank you @lkuligin ! 
Let me know if you have any suggestions.

**Comment by lkuligin** [comment]
@lgesuellip please, fix the remaining linter errors

you can run linter locally if you wish, as described here:
https://github.com/langchain-ai/langchain-google?tab=readme-ov-file#code-formatting

**Review by lkuligin — APPROVED** [review]

**Comment by lkuligin** [comment]
why is `test_vertex_ai_search.py` failing?

**Comment by lgesuellip** [comment]
I just modified it, @lkuligin, and I was able to run the makefile locally to verify it.

**Comment by lgesuellip** [comment]
Hi @lkuligin , do I need to complete any additional steps before merging to main? 
Thank you very much!

**Comment by fpingham** [comment]
hi @lkuligin, all the tests are passing (we also fixed a failing test that was unrelated to our PR)

ready to merge when you are!

**Review by lkuligin — APPROVED** [review]

**Pull request merged**
